client/src/pages/Courses/editables/SemesterField.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import { Flex, Text, Stack, Checkbox } from "@chakra-ui/core";
+import { colors as c } from "../../../colors";
+
+const semMapping = ["S1", "S2", "SS"];
+
+const SemesterField = ({ value, isEditing, onChange }) => {
+    const [checkedSems, setCheckedSems] = useState([]);
+
+    useEffect(() => {
+        setCheckedSems([value.includes("S1"), value.includes("S2"), value.includes("SS")]);
+    }, [value]);
+
+    const updateSem = (i) => {
+        const sems = checkedSems;
+        sems[i] = !sems[i]; // Check/uncheck the semester
+
+        const semesters = [];
+
+        // Determine which semester strings are checked
+        for (let j = 0; j < 3; j++) {
+            sems[j] && semesters.push(semMapping[j]);
+        }
```

Comment:
I don't really like for loops in javascript, WDYT of something like this? 

```suggestion
        // Determine which semester strings are checked
        sems.map((sem, idx) => {
            sem && semesters.push(semMapping[idx]);
        });
```

Reply:
Yea this looks a lot better thanks